src/jinja2_fragments/__init__.py

```diff
@@ -1,10 +1,16 @@
 import typing
+from dataclasses import dataclass
 
 from jinja2 import Environment
 
 
+@dataclass(eq=False)
 class BlockNotFoundError(Exception):
-    pass
+    block_name: str
+    template_name: str
+
+    def __str__(self):
+        return f"Block {self.block_name!r} not found on template {self.template_name!r}"
```

Comment:
I don't think we should set a default message. I would leave that to the raiser of the exception, as it's common practice.

Following a pattern similar to `TemplateNotFound` from jinja, what do you think of this?

```python
class BlockNotFoundError(Exception):
    def __init__(self, block_name: str, template_name: str, message: str | None = None):
        self.block_name = block_name
        self.template_name = template_name
        self.message = message

    def __str__(self):
        return str(self.message)
```

Reply:
I can see some value in letting others use a different message, but falling back to a common default template would be a good practice. I will update the PR accordingly and add a test for it.

Usually, importing exception classes from packages is more commonly done to catch them than raise them. I am unsure about why Jinja's TemplateNotFound is defined like it is. Its code implementing an optional different message was added [more than 13 years ago](https://github.com/pallets/jinja/commit/31bbd9e34d2c612686bb9cebe4efe4e54d81c751) and is no longer used within all of Jinja or Flask's code bases.

Reply:
I really liked your new proposal of keeping it as an input argument and defaulting to a pre-set message if not passed!